web/src/features/datasets/server/service.ts

```diff
@@ -71,8 +70,8 @@ export const createDatasetRunsTableWithoutMetrics = async (
     createdAt: run.run_created_at,
     updatedAt: run.run_updated_at,
     // return metric fields as undefined
-    avgTotalCost: undefined,
-    avgLatency: undefined,
+    avgTotalCost: undefined as Decimal | undefined,
```

Comment:
This makes the frontend type parsing happy and shouldn't make a difference here.